server/src/session.ts

```diff
@@ -386,22 +381,34 @@ export class Session {
     return results;
   }
 
-  private onHover(params: lsp.TextDocumentPositionParams) {
+  private getLSPositionInfo(params: lsp.TextDocumentPositionParams):
```

Comment:
I would prefer either just returning a `scriptInfo` from this function and having the caller create a language service and getting the offset, or rename the function to something like `getLSAndScriptPosition`.